kolide/launcher
# Add kolide_xfconf table for devices using xfce

Closes https://github.com/kolide/launcher/issues/928.

<details>
<summary>Example output</summary>

Invalid username:

```
osquery> select * from kolide_xfconf where username = "very fake";
Error: error generating table: finding user by username 'very fake': user: unknown user very fake
```

Get lock screen settings:

```
osquery> select * from kolide_xfconf where username = 'headless' and fullkey = 'channel/xfce4-power-manager/xfce4-power-manager/lock-screen-suspend-hibernate';
+-------------------------------------------------------------------------------+-------------------------------------------------+-------------------------------+-------+-------+----------+---------------------------------------------------------------------------------+
| fullkey                                                                       | parent                                          | key                           | value | query | username | path                                                                            |
+-------------------------------------------------------------------------------+-------------------------------------------------+-------------------------------+-------+-------+----------+---------------------------------------------------------------------------------+
| channel/xfce4-power-manager/xfce4-power-manager/lock-screen-suspend-hibernate | channel/xfce4-power-manager/xfce4-power-manager | lock-screen-suspend-hibernate | true  | *     | headless | /home/headless/.config/xfce4/xfconf/xfce-perchannel-xml/xfce4-power-manager.xml |
+-------------------------------------------------------------------------------+-------------------------------------------------+-------------------------------+-------+-------+----------+---------------------------------------------------------------------------------+
```

Get autorun settings:

```
osquery> select * from kolide_xfconf where username = 'headless' and fullkey like 'channel/thunar-volman%';
+------------------------------------------------+----------------------------------------+---------+-------+-------+----------+-------------------------------------------------------------+
| fullkey                                        | parent                                 | key     | value | query | username | path                                                        |
+------------------------------------------------+----------------------------------------+---------+-------+-------+----------+-------------------------------------------------------------+
| channel/thunar-volman/automount-media/enabled  | channel/thunar-volman/automount-media  | enabled | false | *     | headless | /etc/xdg/xfce4/xfconf/xfce-perchannel-xml/thunar-volman.xml |
| channel/thunar-volman/automount-drives/enabled | channel/thunar-volman/automount-drives | enabled | false | *     | headless | /etc/xdg/xfce4/xfconf/xfce-perchannel-xml/thunar-volman.xml |
| channel/thunar-volman/autobrowse/enabled       | channel/thunar-volman/autobrowse       | enabled | false | *     | headless | /etc/xdg/xfce4/xfconf/xfce-perchannel-xml/thunar-volman.xml |
| channel/thunar-volman/autoopen/enabled         | channel/thunar-volman/autoopen         | enabled | false | *     | headless | /etc/xdg/xfce4/xfconf/xfce-perchannel-xml/thunar-volman.xml |
+------------------------------------------------+----------------------------------------+---------+-------+-------+----------+-------------------------------------------------------------+
```

Use query to fetch auto-open settings for removable media:
```
osquery> select * from kolide_xfconf where username = 'headless' and query = '*/autoopen*';
+----------------------------------------+--------------------------------+---------+-------+-------------+----------+-------------------------------------------------------------+
| fullkey                                | parent                         | key     | value | query       | username | path                                                        |
+----------------------------------------+--------------------------------+---------+-------+-------------+----------+-------------------------------------------------------------+
| channel/thunar-volman/autoopen/enabled | channel/thunar-volman/autoopen | enabled | false | */autoopen* | headless | /etc/xdg/xfce4/xfconf/xfce-perchannel-xml/thunar-volman.xml |
+----------------------------------------+--------------------------------+---------+-------+-------------+----------+-------------------------------------------------------------+
```
</details>

<details>

<summary>
Testing steps
</summary>

* Use image https://hub.docker.com/r/accetto/ubuntu-vnc-xfce-g3 -- `docker run --platform linux/amd64 -it --name accettovnc -p 25901:5901 -p 26901:6901 accetto/ubuntu-vnc-xfce-g3:latest`
* Install xfce4-power-manager, thunar-volman, and dbus-x11; run `export $(dbus-launch)`; install osquery; copy built binary into container
* Over novnc (http://localhost:26901/vnc.html) configure some power and autorun settings in the GUI to query later
* In the container shell, run launcher interactive; new table `kolide_xfconf` should be listed in schema
* Query by username and optionally channel/key

</details>
